src/Functions/FunctionsConversion.cpp

```diff
@@ -4081,23 +4081,43 @@ class FunctionCast final : public IFunctionBase
             "Cast to Object can be performed only from flatten named Tuple, Map or String. Got: {}", from_type->getName());
     }
 
+
     WrapperType createObjectWrapper(const DataTypePtr & from_type, const DataTypeObject * to_object) const
     {
         if (checkAndGetDataType<DataTypeString>(from_type.get()))
         {
             return [this](ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, const ColumnNullable * nullable_source, size_t input_rows_count)
             {
-                auto res = ConvertImplGenericFromString<true>::execute(arguments, result_type, nullable_source, input_rows_count, context)->assumeMutable();
-                res->finalize();
-                return res;
+                return ConvertImplGenericFromString<true>::execute(arguments, result_type, nullable_source, input_rows_count, context);
```

Comment:
I do not actually understand why you removed finalize here, but if it is ok, I believe you

Reply:
`finalize` makes sense only for columns with old `Object` data type. Here it's unneded and was added only due to bad copy-paste actually